src/components/App/App.jsx

```diff
@@ -0,0 +1,400 @@
+    const token = getToken();
+    if (!token) return;
+
+    // Find the matching article in savedArticles by comparing URLs
+    const savedArticle = savedArticles.find(
+      (saved) => saved.url === article.url
+    );
+
+    if (savedArticle) {
+      const articleId = savedArticle._id; // Use MongoDB _id from savedArticles
+
+      // Call the backend to delete the article by its _id
+      auth
+        .deleteArticle(articleId, token) // API call to delete the article by _id
+        .then(() => {
+          console.log("Article deleted:", articleId);
+          // Remove the article from savedArticles state
+          setSavedArticles((prevArticles) =>
+            prevArticles.filter((a) => a._id !== articleId)
+          );
+        })
+        .catch((error) => {
+          console.error("Error deleting article:", error);
+        });
+    } else {
+      console.log("No matching saved article found for deletion.");
+    }
+  };
+
+  const handleSignin = () => {
+    setActiveModal("sign-in");
+  };
+
+  const handleSignup = () => {
+    setActiveModal("sign-up");
+  };
+
+  const closeActiveModal = () => {
+    setActiveModal("");
+  };
+
+  const handleOpenModal = () => {
+    setIsModalOpen(true);
+  };
+
+  useEffect(() => {
+    const handleKeyDown = (e) => {
+      if (e.key === "Escape" && activeModal) {
+        closeActiveModal();
+      }
+    };
+
+    document.addEventListener("keydown", handleKeyDown);
+
+    return () => {
+      document.removeEventListener("keydown", handleKeyDown);
+    };
+  }, [activeModal]);
+
+  const handleOutsideClick = (e) => {
+    if (e.target.classList.contains("modal_opened")) {
+      closeActiveModal();
+    }
+  };
+
+  // Handle search logic
+  const handleSearch = async (searchQuery) => {
+    if (!searchQuery.trim()) {
+      setError("Please enter a keyword");
+      return;
+    }
+
+    setIsLoading(true);
+    setError(null);
+    setIsSubmitted(true);
+    setSearchQuery(searchQuery);
+
+    console.log("Search query captured:", searchQuery);
+
+    try {
+      const news = await getNews(searchQuery);
+
+      console.log("Fetched articles for query:", news); // Log fetched articles to ensure data is coming through
+
+      setArticles(news);
+      if (news.length === 0) {
+        setError("No articles found for this query.");
+      }
+    } catch (err) {
+      setError("Sorry, something went wrong during the request.");
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  return (
+    <>
+      <div className="page">
+        <div className="page__content">
+          <div className="page__style">
+            <Header
+              handleSignIn={handleSignin}
+              handleSignup={handleSignup}
+              onClose={closeActiveModal}
+              isOpen={activeModal === "sign-in"}
+              isModalOpen={isModalOpen}
+              isLoggedIn={isLoggedIn}
+              handleLogout={handleLogout}
+              userData={userData}
```

Comment:
`userData={userData}` is not needed in the props because you should use `CurrentUserContext`